core/client/index.ts

```diff
@@ -0,0 +1,109 @@
+/*
+ ************
+ ************
+ *** Core ***
+ ************
+ ************
+ */
+
+import EventEmitter from "events";
+import { OlmWrapper } from  "./olmWrapper.js";
+import { ServerComm } from "./serverComm.js";
+import { payloadType } from "../../higher";
+
+export type outboundEncPayloadType = {
+  deviceId: string,
+  payload: string
+};
+
+export type inboundEncPayloadType = {
+  seqID: number,
+  sender: string,
+  encPayload: string,
+};
+
+export class Core {
+  olmWrapper: OlmWrapper;
+  #serverComm: ServerComm;
+  eventEmitter: EventEmitter;
+
+  /**
+   * Initializes client-server connection and client state.
+   *
+   * @param {string} ip server IP address
+   * @param {string} port server port number
+   * @param {{ onAuth: callback,
+   *           onUnauth: callback, 
+   *           validateCallback: callback}} config client configuration options
+   */
+  constructor(
+      eventEmitter: EventEmitter,
+      turnEncryptionOff: boolean,
+      ip?: string,
+      port?: string) {
+    this.olmWrapper = new OlmWrapper(turnEncryptionOff);
+    this.eventEmitter = eventEmitter;
+    // register listener for incoming messages
+    this.eventEmitter.on('serverMsg', async (msg) => {
+      await this.onMessage(msg);
+    });
+    this.#serverComm = new ServerComm(this.eventEmitter, ip, port);
+  }
+
+  async init() {
```

Comment:
If other methods have an invariant that `init` was called, it's probably only expose a constructor that initializes it. I think you can do this with a private constructor and static factory.

So, for example, I might have:

```ts
class Core {
  private constructor(eventEmitter, turnEncryptionOff, ...) { ... }
  async init(..) { ... }
  public static async create(eventEmitter, turnEncryptionOff,...) {
    let result = new Core(eventEmitter, ...);
    await result.init();
    return result;
  }
}
```